internal/consensus/state.go

```diff
@@ -2686,3 +2712,8 @@ func proposerWaitTime(lt cmttime.Source, bt time.Time) time.Duration {
 	}
 	return 0
 }
+
+// isPBTSEnabled returns true if PBTS is enabled at the current height.
```

Comment:
We have the lock when using this, haven't we? If so, probably we should write a "contract" comment.

Reply:
This is true for `IsPBTSEnabled` (except under testing).
Not true for `state.ConsensusParams.Feature.PbtsEnabled(height)`, which is also called in other places (under testing and when performing block sync). This is actually causing a race caught in #2335

So I propose to treat this in the context of #2335 as well